### PR TITLE
Implement FE_WedgeDGP and FE_PyramidDGP

### PR DESCRIPTION
**Pull request by peterrum**
... and introduce the base classes `FE_Wedge` and `FE_Pyramid`.

### CODE REVIEW

File: include/deal.II/simplex/fe_lib.h [hunk 1]
```diff
@@ -229,16 +247,65 @@ namespace Simplex
...
+     */
+    FE_Pyramid(const unsigned int                                degree,
+               const internal::GenericDoFsPerObject &            dpos,
+               const typename FiniteElementData<dim>::Conformity conformity);
+  };
 
   /**
    * Implementation of a scalar Lagrange finite element on a pyramid that yields
-   * the finite element space of continuous, piecewise polynomials of
+   * the finite element space of discontinuous, piecewise polynomials of
```
Comment (drwells) [inline include/deal.II/simplex/fe_lib.h:302]:
```suggestion
   * the finite element space of continuous, piecewise polynomials of
```

File: include/deal.II/simplex/fe_lib.h [hunk 2]
```diff
@@ -289,6 +356,37 @@ namespace Simplex
                            const unsigned int face_no = 0) const override;
   };
 
+  /**
+   * Implementation of a scalar Lagrange finite element on a pyramid that yields
+   * the finite element space of continuous, piecewise polynomials of
```
Comment (drwells) [inline include/deal.II/simplex/fe_lib.h:361]:
```suggestion
   * the finite element space of discontinuous, piecewise polynomials of
```

File: include/deal.II/fe/fe_base.h [hunk 3]
```diff
@@ -673,6 +673,14 @@ class FiniteElementData
   get_first_face_quad_index(const unsigned int face_no = 0) const;
 };
 
+namespace internal
+{
+  internal::GenericDoFsPerObject
```
Comment (drwells) [inline include/deal.II/fe/fe_base.h:678]: Could we document this function?

**Review by drwells — CHANGES_REQUESTED** [review]

**Comment by peterrum** [comment]
@drwells Done!

**Review by drwells — APPROVED** [review]

**Pull request merged**